src/main/starlark/builtins_bzl/common/cc/experimental_cc_shared_library.bzl

```diff
@@ -72,16 +107,45 @@ def _separate_static_and_dynamic_link_libraries(
             break
 
         node = all_children[i]
-        node_label = str(node.label)
-
-        if node_label in seen_labels:
-            continue
-        seen_labels[node_label] = True
 
-        if node_label in can_be_linked_dynamically:
-            targets_to_be_linked_dynamically_set[node_label] = True
-        else:
-            targets_to_be_linked_statically_map[node_label] = node.linkable_more_than_once
+        must_add_children = False
+        # The seen count is used to track a programmatic error and fail if it
```

Comment:
nit: Add _ to match the variable name, this is more consistent with the rest of the comment.

Reply:
Done.